onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.h

```diff
@@ -338,6 +339,7 @@ struct ShapeInformationAdapter : IShapeInformationAdapter
 
     virtual uint32_t GetInputTensorDimensionCount(uint32_t inputIndex) const { return m_informationSource.GetInputTensorDimensionCount(inputIndex); }
     virtual std::vector<uint32_t> GetInputTensorShape(uint32_t inputIndex) const { return m_informationSource.GetInputTensorShape(inputIndex); }
+    virtual ~ShapeInformationAdapter() {}
```

Comment:
Should the destructor of IShapeInformationAdapter be virtual? I'm not familiar with COM and I'm not sure if it is possible.

Reply:
Done.